fdbserver/storageserver.actor.cpp

```diff
@@ -10379,15 +10379,14 @@ void changeServerKeysWithPhysicalShards(StorageServer* data,
 				} else {
 					ASSERT(shard->adding != nullptr || shard->moveInShard != nullptr);
 					if (shard->desiredShardId != desiredId) {
-						TraceEvent(SevError, "CSKConflictingMoveInShards", data->thisServerID)
+						TraceEvent(SevWarnAlways, "CSKConflictingMoveInShards", data->thisServerID)
 						    .detail("DataMoveID", dataMoveId)
 						    .detail("Range", range)
 						    .detailf("TargetShard", "%016llx", desiredId)
 						    .detailf("CurrentShard", "%016llx", shard->desiredShardId)
 						    .detail("IsTSS", data->isTss())
 						    .detail("Version", cVer);
-						// TODO(heliu): Mark the data move as failed locally, instead of crashing ss.
-						ASSERT(false);
+						throw data_move_conflict();
```

Comment:
Seems that this error can fail the storageServerCore, do we want to handle it when update() captures this error? Do we want to distinguish the error handling for SS vs. TSS?

Reply:
This error should not happen for regular storage server and retry this action would block storage server from pulling new mutations. For TSS, I believe it's better to have it fail explicitly rather than running with an inconsistent state.